resources/conditions/kube_lease.go

```diff
@@ -14,6 +14,7 @@ import (
 const (
 	PodNameEnvar           = "POD_NAME"
 	DefaultLeaderLeaseName = "elastic-agent-cluster-leader"
+	DefaultLeaderValue     = true
```

Comment:
It seems like the default value here may remain false.
If we create the cluster properly with `cloudbeat-ds.yaml` the entire flow should work as expected without fallbacks.
Did I miss anything?
If that's the case, let's rename the ticket to a more indicative name.

Reply:
reverting back to true, still keeping the changes so it'll make it easy for us to change default value locally